deps/chakrashim/src/v8objecttemplate.cc

```diff
@@ -868,6 +869,25 @@ Local<Object> ObjectTemplate::NewInstance(Handle<Object> prototype) {
     return Local<Object>();
   }
 
+  if (prototype.IsEmpty()) {
+    // Try the constructor function, if it exists
+    if (!objectTemplateData->constructor.IsEmpty()) {
+      Local<Function> function = objectTemplateData->constructor->GetFunction();
+
+      if (!function.IsEmpty()) {
+        jsrt::IsolateShim* iso = jsrt::IsolateShim::GetCurrent();
+        JsValueRef prototypeValue = nullptr;
+
+        if (JsGetProperty(*function,
+                          iso->GetCachedPropertyIdRef(
+                              jsrt::CachedPropertyIdRef::prototype),
+                          &prototypeValue) == JsNoError) {
+          prototype = Local<Object>::New(prototypeValue);
+        }
+      }
+    }
+  }
+
   if (!prototype.IsEmpty()) {
```

Comment:
Is there a reason you chose not to make this an `else`?

I'd also like to see a comment here explaining the scenario this is supporting; it's something that I could easily see being broken if it is forgotten.

Reply:
The reason is that the previous conditional is just an alternate way to get the prototype.  It could have already been provided directly.

Reply:
Ah, I'd missed that it was setting `prototype`